src/metabase/api/database.clj

```diff
@@ -1112,8 +1113,8 @@
    (schema-tables-list db-id schema nil nil))
   ([db-id schema include_hidden include_editable_data_model]
    (when-not include_editable_data_model
-     (api/read-check Database db-id))
-   (api/check-403 (can-read-schema? db-id schema))
+     (api/read-check Database db-id)
+     (api/check-403 (can-read-schema? db-id schema)))
```

Comment:
Same as above. This fixes a bug in the feature branch. We were checking for data permissions but we should only check for data-model permissions if include_editable_data_model=true